pkg/services/folder/folderimpl/folder.go

```diff
@@ -298,13 +299,14 @@ func (s *Service) GetChildren(ctx context.Context, q *folder.GetChildrenQuery) (
 		return nil, err
 	}
 
-	canView, err := g.CanView()
-	if err != nil {
-		return nil, err
+	guardianFunc := g.CanView
+	if q.Permission == dashboardaccess.PERMISSION_EDIT {
+		guardianFunc = g.CanEdit
```

Comment:
not sure if we want `CanEdit()` or `CanSave()` here
I wonder if there is any difference for folders

Reply:
Apparently, CanSave means user has write permission to the containing folder (i.e. can save a new dashboard, aka "create"). CanEdit means a user has a Write permission to that very dashboard.